src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java

```diff
@@ -452,9 +452,19 @@ public static Map<String, DetailNode> getPropertiesJavadocs(Set<String> properti
         final Map<String, DetailNode> unmodifiableJavadocs =
                 ClassAndPropertiesSettersJavadocScraper.getJavadocsForModuleOrProperty();
         final Map<String, DetailNode> javadocs = new LinkedHashMap<>(unmodifiableJavadocs);
-        properties.forEach(property -> {
+
+        for (String property : properties) {
             javadocs.putIfAbsent(property, SUPER_CLASS_PROPERTIES_JAVADOCS.get(property));
-        });
+
+            if (javadocs.get(property) == null
+                    && !TOKENS.equals(property)
+                    && !JAVADOC_TOKENS.equals(property)) {
```

Comment:
Please extract logic to some descriptively named variable. `isMissingPropertyJavadoc`?

Reply:
!isPropertyJavadocScraped  ?

Reply:
On second thought,

>Failed to find javadoc for property 'query'

shouldn't this be `Failed to find setter javadoc for property 'query'`?

So `isPropertySetterJavadocFound`?

Reply:
Done.